src/sentry/search/eap/utils.py

```diff
@@ -33,6 +33,14 @@ def _validator(input: str) -> bool:
     return _validator
 
 
+def number_validator(input: str) -> bool:
+    try:
+        float(input)
```

Comment:
I'm not a fan of using exceptions for flow control, I'd much prefer if we either use a regex or do `input.replace('.', '', 1).isdecimal()`

Reply:
Fair! Updated to use `input.replace`!